### PR TITLE
eth/polygon: Add USDT

### PR DESCRIPTION
**Pull request by martonp**

### CODE REVIEW

**Comment by exitus1** [comment]
Got a question: Does this indicate USDT will be tradable stablecoin in the future?

File: client/webserver/site/src/img/coins/usdt.png
Comment (buck54321) [inline client/webserver/site/src/img/coins/usdt.png:1]: 100 x 100 px please

File: dex/networks/eth/tokens.go [hunk 1]
```diff
@@ -173,20 +173,119 @@ var Tokens = map[uint32]*Token{
 			},
 		},
 	},
+	usdtTokenID: {
+		EVMFactor: new(int64),
+		Token: &dex.Token{
+			ParentID: EthBipID,
+			Name:     "USDT",
```
Comment (buck54321) [inline dex/networks/eth/tokens.go:180]: Probably "Tether"

File: dex/networks/eth/tokens.go [hunk 2]
```diff
@@ -173,20 +173,119 @@ var Tokens = map[uint32]*Token{
...
+		NetTokens: map[dex.Network]*NetToken{
+			dex.Mainnet: {
+				Address: common.HexToAddress("0xdac17f958d2ee523a2206206994597c13d831ec7"), // https://etherscan.io/address/0xdac17f958d2ee523a2206206994597c13d831ec7
+				SwapContracts: map[uint32]*SwapContract{
+					0: {
+						// swap contract: https://etherscan.io/address/0x97a53fEF7854f4CB846F2eaCCf847229F1E10e4f
+						Address: common.HexToAddress("0x97a53fEF7854f4CB846F2eaCCf847229F1E10e4f"),
+						// USDT's contract is upgradable, using a proxy call, so
+						// gas cost could change without notice, so we do not
+						// want to set limits too low, even with live estimates.
```
Comment (buck54321) [inline dex/networks/eth/tokens.go:198]: It looks like you're just using 30% buffer from getgas. Are you suggesting we should add more?

File: dex/testing/dcrdex/genmarkets.sh [hunk 3]
```diff
@@ -148,6 +148,24 @@ if [ $ETH_ON -eq 0 ]; then
...
+            "parcelSize": 4
+        },
+        {
+            "base": "DCR_simnet",
+            "quote": "USDT_simnet",
+            "lotSize": 100000000,
+            "rateStep": 100000,
+            "epochDuration": ${EPOCH_DURATION},
+            "marketBuyBuffer": 1.2,
+            "parcelSize": 4
```
Comment (buck54321) [inline dex/testing/dcrdex/genmarkets.sh:168]: Is one enough? Or make the second one a pair with a token?

**Review by buck54321 — CHANGES_REQUESTED** [review]

File: dex/networks/eth/tokens.go (hunk 2 above)
Comment (martonp) [reply dex/networks/eth/tokens.go:198]: No, I think it's fine. This comment was just copied over from USDC.

**Review by JoeGruffins — APPROVED** [review]

**Review by buck54321 — APPROVED** [review]

**Pull request merged**